Userland/Libraries/LibGUI/VimEditingEngine.cpp

```diff
@@ -1411,4 +1411,12 @@ void VimEditingEngine::move_to_next_empty_lines_block()
     m_editor->set_cursor(new_cursor);
 };
 
+void VimEditingEngine::uppercase_selection()
+{
+    VERIFY(!m_editor.is_null());
+    VERIFY(m_editor->has_selection());
+
+    m_editor->insert_at_cursor_or_replace_selection(m_yank_buffer.to_uppercase());
```

Comment:
why does this function expect you to have already yanked the selection. Also this will overwrite the yank buffer, which is not what vim does. why not use `m_editor->selected_text()` which yank uses internally.

Reply:
Oh, nice. This is what I was trying to do actually but missed the m_editor->selected_text() function and thought the yank buffer was the way to go. Fixed!